client/src/sections/resources/IndustrialFireExtinguisher.tsx

```diff
@@ -0,0 +1,97 @@
+import Line from "../../components/Line";
+import Tile from "../../components/Tile";
+import useHave from "../../hooks/useHave";
+import useGet from "../../hooks/useGet";
+import { Step, useQuestStep } from "../../hooks/useQuest";
+import { $item } from "../../util/makeValue";
+
+const IndustrialFireExtinguisher = () => {
+    const foam = useGet("_fireExtinguisherCharge");
+
+    const knobUnfinished = useQuestStep("questL05Goblin") < Step.FINISHED;
```

Comment:
just wanted to poke in and say i like the "unfinished, unused, queststep, broader conditional" structure you used for each of these charges. good job.